### PR TITLE
Remove callback directive

### PR DESCRIPTION
**Pull request by mjfwebb**
# Description

This PR removes the deprecated `@callback` directive. Both the code itself and the documentation for the directive have been removed.

Tests using `@callback` have been updated to use `@populatedBy`.

## Complexity

Low

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
| 🟥 query.QueryWithNestedIn | 15127 | 13328 | 52 | 72 | 1794 |


🟥 - Performance worsened (dbHits)
🟩 - Performance improved (dbHits)
🟦 - New test
        

<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 31 | 39 | 1134 |
|  aggregations.NestedAggregation | 14551 | 14551 | 86 | 119 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 64 | 97 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 2748 | 2992 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2738 | 2772 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 53 | 93 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 79 | 95 | 2174 |
|  aggregations.DeeplyNestedCount | 13070327 | 13070331 | 4392 | 4323 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 128 | 151 | 600 |
|  connect.createAndConnect | 14424 | 14424 | 140 | 199 | 3003 |
|  connections.Connection | 14082 | 14082 | 68 | 79 | 2174 |
|  connections.NestedConnection | 41462 | 41466 | 133 | 144 | 4516 |
|  connections.ConnectionWithSort | 2282 | 2282 | 78 | 63 | 1040 |
|  connections.ConnectionWithSortAndCypher | 15323 | 15323 | 85 | 141 | 2174 |
|  create.SimpleMutation | 7 | 7 | 52 | 69 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 31 | 38 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 242 | 247 | 1040 |
|  delete.NestedDeleteInUpdate | 18196 | 18196 | 228 | 260 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 37 | 46 | 1040 |
|  query.SimpleQuery | 15160 | 15160 | 70 | 70 | 2174 |
|  query.QueryWhere | 9698 | 9715 | 34 | 53 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9876 | 9893 | 49 | 58 | 2161 |
|  query.Nested | 10092037 | 10092037 | 10425 | 10668 | 2008534 |
|  query.NestedWithFilter | 40401 | 40401 | 103 | 127 | 4511 |
|  query.OrFilterOnRelationships | 43441 | 43517 | 156 | 208 | 1898 |
|  query.OrFilterOnRelationshipsAndNested | 36695 | 37052 | 210 | 279 | 1898 |
| 🟥 query.QueryWithNestedIn | 15127 | 13328 | 52 | 72 | 1794 |
|  query.NestedConnectionWhere | 9834 | 9834 | 72 | 94 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9992 | 9834 | 68 | 110 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6343 | 6343 | 114 | 176 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3024400 | 3024400 | 536 | 569 | 1003942 |
|  query.Fulltext | 96 | 96 | 24 | 27 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 40 | 45 | 84 |
|  sorting.SortMultipleTypes | 2513 | 2513 | 60 | 68 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14495 | 14472 | 86 | 106 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 126 | 95 | 2174 |
|  sorting.SortDeeplyNestedFields | 42196 | 42196 | 115 | 116 | 4516 |
|  sorting.SortWithTopLevelCypher | 15160 | 15160 | 51 | 126 | 2174 |
|  unions.SimpleUnionQuery | 321 | 321 | 80 | 95 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 68 | 89 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 313 | 322 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 290 | 316 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 65 | 87 | 1040 |

</details>
        
Old Schema Generation: 27.556s
Schema Generation: 27.474s
Old Subgraph Schema Generation: 40.938s
Subgraph Schema Generation: 41.236s

File: packages/graphql/tests/integration/deprecated/1414.int.test.ts
Comment (angrykoala) [inline packages/graphql/tests/integration/deprecated/1414.int.test.ts:1]: This is part of the `deperecated` tests and it is updated already with `populatedBy` outside this folder. Now that `callback` is dropped we can safely removed the `deprecated` tests related to it

**Review by angrykoala — CHANGES_REQUESTED** [review]
The changes look good, just make sure to remove all the deprecated tests related to `callback`

File: packages/graphql/tests/integration/deprecated/1414.int.test.ts
Comment (mjfwebb) [reply packages/graphql/tests/integration/deprecated/1414.int.test.ts:1]: Addressed in https://github.com/neo4j/graphql/pull/3062/commits/c99941ba95115c2a2990fe04c7f9f01be6b8e3d8 :+1:

**Pull request merged**